mirascope/tools/system/_computer_use.py

```diff
@@ -0,0 +1,153 @@
+from __future__ import annotations
+
+import io
+import tarfile
+from abc import ABC
+from contextlib import suppress
+from typing import ClassVar
+
+import docker
+from docker.models.containers import Container
+from pydantic import Field
+
+from mirascope.core import toolkit_tool
+from mirascope.tools.base import (
+    ConfigurableTool,
+    ConfigurableToolKit,
+    _ToolConfig,
+    _ToolSchemaT,
+)
+
+
+class ComputerUseToolkitConfig(_ToolConfig):
+    """Configuration for computer use toolkit"""
+
+    docker_image: str = Field(
+        default="python:3.13-slim", description="Docker image for code execution"
+    )
+    max_memory: str = Field(default="512m", description="Maximum memory allocation")
+    allow_network: bool = Field(default=True, description="Allow network access")
+
+
+class DockerOperation(ConfigurableTool[ComputerUseToolkitConfig, _ToolSchemaT], ABC):
+    """Base class for Docker operations."""
+
+    __config__ = ComputerUseToolkitConfig()
```

Comment:
hmm, so each of these tools has a different config, and it's not the same as the parent toolkit's configuration? It seems like each tool should have it's own `_ConfigurableToolConfig` if necessary? How does the user configure each individual tool? Is that even useful?

More so I care about figuring out how to propagate the toolkit's config down into the tools that are part of the toolkit. This way the user could set configuration options for the toolkit and each tool can access that same config.

This is also inline with `__config__` not making sense as a class variable (see other comment)

Reply:
When from_config is called, the contents of the config are communicated to `tool`.
The problem is that it is difficult to access higher level classes due to the placement of classes within classes. Let's think about it some more.a